tokio/src/io/async_fd.rs

```diff
@@ -1257,3 +1313,43 @@ impl<'a, T: std::fmt::Debug + AsRawFd> std::fmt::Debug for AsyncFdReadyMutGuard<
 /// [`try_io`]: method@AsyncFdReadyGuard::try_io
 #[derive(Debug)]
 pub struct TryIoError(());
+
+/// Error returned by [`try_new`] or [`try_with_interest`].
+///
+/// [`try_new`]: AsyncFd::try_new
+/// [`try_with_interest`]: AsyncFd::try_with_interest
+pub struct AsyncFdError<T> {
```

Comment:
If we're going to define a new type for this, then we should probably call it `AsyncFdTryNewError`.

But I would also be okay with `(T, io::Error)`.